components/service/glean/src/main/java/mozilla/components/service/glean/scheduler/MetricsPingScheduler.kt

```diff
@@ -224,23 +223,11 @@ internal class MetricsPingScheduler(val applicationContext: Context) {
         // this are sent in the next window.  If we are in the background, then we may stay there
         // until the app is killed so we shouldn't reschedule unless the app is foregrounded again
         // (see GleanLifecycleObserver).
-        if (isApplicationInForeground()) {
+        if (GleanLifecycleObserver.isInForeground) {
```

Comment:
Instead of making `MetricsPingScheduler` tightly coupled to `GleanLifecycleObserver`, what about it provides a `setForegroundStatus(isForeground: boolean)` and we call it from `GleanLifecycleObserver`? Another option would be to have another observer embedded in the scheduler, to make it completely stand-alone. But I have no strong opinion on this last bit.

Moreover, by providing the method, you might be able to write tests more easily.